File: zero-business/zero-main-business/zero-main-domain-business/src/main/scala/com/zjj/main/domain/role/RoleRepository.scala
package com.zjj.main.domain.role

import java.lang

/**
 * @author zengJiaJun
 * @crateTime 2024年11月21日 21:12
 * @version 1.0
 */
trait RoleRepository {

  def findById(id: lang.Long): io.vavr.control.Option[RoleAgg]
  def findByRoleKey(id: String): io.vavr.control.Option[RoleAgg]

  def save(roleAgg: RoleAgg): Unit
}

File: zero-business/zero-main-business/zero-main-domain-business/src/main/java/com/zjj/main/domain/role/RoleRepository.java
package com.zjj.main.domain.role;

import io.vavr.control.Option;

import java.util.List;
import java.util.Set;

/**
 * @author zengJiaJun
 * @version 1.0
 * @crateTime 2025年03月24日 16:11
 */
public interface RoleRepository {
    Option<RoleAgg> findById(Long id);
    Option<RoleAgg> findByRoleKey(String roleKey);
    void save(RoleAgg roleAgg);

    List<Role> findAllByIdIn(Set<Long> ids);

    boolean existsByCode(String code);
}
